src/Aspire.Hosting/Orchestrator/ApplicationOrchestrator.cs

```diff
@@ -238,6 +238,11 @@ await _notificationService.PublishUpdateAsync(child, s => s with
                 StopTimeStamp = stopTimeStamp,
                 Properties = s.Properties.SetResourceProperty(KnownProperties.Resource.ParentName, parentName)
             }).ConfigureAwait(false);
+
+            // the parent name needs to be an instance name, not the resource name.
+            // parent the children of the child under the first resource instance.
+            await SetChildResourceAsync(child, child.GetResolvedResourceNames()[0], state, startTimeStamp, stopTimeStamp)
```

Comment:
This isn't going to work well with replicas. But that isn't new.

Reply:
How would you expect it to work with replicas? Which replica should the child go under?

Reply:
I don't think it can work today. Lets see if it comes up before worrying about it.